File: web/src/utils/fileKind.js
/** Classify project/main upload for preview + cards (web + shared logic). */
export function fileKind(url, mime = '', name = '') {
  const hay = `${url || ''} ${name || ''}`.toLowerCase();
  const m = (mime || '').toLowerCase();

  if (m.includes('pdf') || /\.pdf(\?|#|$)/i.test(hay)) return 'pdf';
  if (
    m.startsWith('image/') ||
    /\.(jpe?g|png|gif|webp|bmp)(\?|#|$)/i.test(hay)
  ) {
    return 'image';
  }
  if (
    m.startsWith('text/') ||
    /\.(txt|csv)(\?|#|$)/i.test(hay)
  ) {
    return 'text';
  }
  if (
    /\.(docx?|pptx?|xlsx?)(\?|#|$)/i.test(hay) ||
    m.includes('officedocument') ||
    m.includes('msword') ||
    m.includes('ms-excel') ||
    m.includes('ms-powerpoint')
  ) {
    return 'doc';
  }
  return 'unknown';
}

export function isPdfFile(url, mime = '', name = '') {
  return fileKind(url, mime, name) === 'pdf';
}

export function isImageFile(url, mime = '', name = '') {
  return fileKind(url, mime, name) === 'image';
}

File: mobile/lib/utils/file_kind.dart
/// Shared file-kind helpers (keep in sync with web/src/utils/fileKind.js).

String _hay(String? url, String? mime, String? name) =>
    '${url ?? ''} ${name ?? ''}'.toLowerCase();

String fileKind(String? url, {String? mime, String? name}) {
  final hay = _hay(url, mime, name);
  final m = (mime ?? '').toLowerCase();

  if (m.contains('pdf') || RegExp(r'\.pdf(\?|#|$)').hasMatch(hay)) {
    return 'pdf';
  }
  if (m.startsWith('image/') ||
      RegExp(r'\.(jpe?g|png|gif|webp|bmp)(\?|#|$)').hasMatch(hay)) {
    return 'image';
  }
  if (RegExp(r'\.(docx?|pptx?|xlsx?|txt|csv)(\?|#|$)').hasMatch(hay) ||
      m.contains('officedocument') ||
      m.contains('msword') ||
      m.contains('ms-excel') ||
      m.contains('ms-powerpoint') ||
      m.contains('text/')) {
    return 'doc';
  }
  if ((url ?? '').contains('/uploads/')) return 'doc';
  return 'unknown';
}

bool isPdfFile(String? url, {String? mime, String? name}) =>
    fileKind(url, mime: mime, name: name) == 'pdf';

bool isImageFile(String? url, {String? mime, String? name}) =>
    fileKind(url, mime: mime, name: name) == 'image';
